From: Bhavna Sarathy <bnagendr@redhat.com>
Date: Thu, 5 Aug 2010 18:04:35 -0400
Subject: [edac] amd64_edac: fix operator precedence error
Message-id: <20100805181015.6566.75639.sendpatchset@localhost.localdomain>
Patchwork-id: 27423
O-Subject: [RHEL5.6 PATCH 9/10] amd64_edac - Fix operator precedence error
Bugzilla: 568576
RH-Acked-by: Mauro Carvalho Chehab <mchehab@redhat.com>

>From 545d36aff8a830ea6bb6b1f6ad7eb61fe3f5b940 Mon Sep 17 00:00:00 2001
From: Borislav Petkov <borislav.petkov@amd.com>
Date: Wed, 4 Aug 2010 02:47:39 +0200
Subject: [PATCH] amd64_edac: Fix operator precendence error

The bitwise AND is higher prio.

Signed-off-by: Borislav Petkov <borislav.petkov@amd.com>

diff --git a/drivers/edac/amd64_edac.c b/drivers/edac/amd64_edac.c
index 5da611b..6edbe02 100644
--- a/drivers/edac/amd64_edac.c
+++ b/drivers/edac/amd64_edac.c
@@ -1446,7 +1446,7 @@ static void f10_map_sysaddr_to_csrow(struct mem_ctl_info *mci,
 	 * ganged. Otherwise @chan should already contain the channel at
 	 * this point.
 	 */
-	if (dct_ganging_enabled(pvt) && pvt->nbcfg & K8_NBCFG_CHIPKILL)
+	if (dct_ganging_enabled(pvt) && (pvt->nbcfg & K8_NBCFG_CHIPKILL))
 		chan = get_channel_from_ecc_syndrome(mci, syndrome);
 
 	if (chan >= 0)
